src/Factory/HalViewHelperFactory.php

```diff
@@ -30,7 +30,8 @@ public function createService(ServiceLocatorInterface $serviceLocator)
         }
         $urlHelper       = $serviceLocator->get('Url');
 
-        $helper = new Plugin\Hal($hydrators);
+        /** @var Plugin\Hal $helper */
+        $helper = $services->get('ControllerPluginManager')->get('Hal');
```

Comment:
@andriesss Changing this means that if you want to override the view helper and controller plugin, you need to override the HAL view helper's factory; at that point, the controller plugin manager will pick up your alternate helper.